src/main/java/xyz/nucleoid/bedwars/game/active/BwActive.java

```diff
@@ -211,7 +204,12 @@ private void addPlayer(ServerPlayerEntity player) {
             this.spawnLogic.spawnAtCenter(player);
         }
     }
-
+    private boolean onTakeDamage(ServerPlayerEntity player, DamageSource source, float v){
+        if (source == DamageSource.OUT_OF_WORLD && player.isSpectator()){
+            return true;
```

Comment:
This could be condensed into a single `return` line.

Reply:
It'd also probably be valid to cancel all damage to spectators- not strictly necessary though. I think the only damage that can apply to spectators is void damage anyway.